gui/src/main/proxy.js

```diff
@@ -0,0 +1,107 @@
+/* eslint standard/no-callback-literal: 0 */
+
+const ElectronProxyAgent = require('electron-proxy-agent')
+const url = require('url')
+const http = require('http')
+const https = require('https')
+
+const log = require('cozy-desktop').default.logger({
+  component: 'GUI:proxy'
+})
+
+const config = require('yargs')
+  .env('COZY_DRIVE')
+  .conflicts('proxy-script', 'proxy-rules')
+  .describe('proxy-script', 'The URL associated with the PAC file.')
+  .describe('proxy-rules', 'Rules indicating which proxies to use.')
+  .describe('proxy-bypassrules', 'Rules indicating which URLs should bypass the proxy settings. ' +
+                                 'See https://github.com/electron/electron/blob/master/docs/api/session.md#sessetproxyconfig-callback')
+  .default('proxy-ntlm-domains', '*')
+  .describe('proxy-ntlm-domains', 'A comma-separated list of servers for which integrated authentication is enabled. ' +
+                                  'Dynamically sets whether to always send credentials for HTTP NTLM or Negotiate authentication.')
+  .describe('login-by-realm', 'comma-separated list of realm:user:password')
+  .help('help')
+  .parse()
+
+log.warn({config}, 'argv')
+
+const printCertificate = (certif) => `Certificate(${certif.issuerName} ${certif.subjectName})`
+
+module.exports = (app, session, doneSetup) => {
+  const loginByRealm = {}
+  if (config['login-by-realm']) {
+    config['login-by-realm'].split(',').forEach((lbr) => {
+      const [realm, username, password] = lbr.split(':')
+      loginByRealm[realm] = [username, password]
+    })
+  }
+
+  if (config['proxy-ntlm-domains']) {
+    session.defaultSession.allowNTLMCredentialsForDomains(config['proxy-ntlm-domains'])
+  }
+
+  session.defaultSession.setCertificateVerifyProc((request, callback) => {
+    const {hostname, certificate, verificationResult, errorCode} = request
+    if (verificationResult < 0) {
+      log.warn({hostname, certificate: printCertificate(certificate), verificationResult, errorCode}, 'Certificate Verification Error')
+    } else {
+      log.debug({hostname, certificate: printCertificate(certificate), verificationResult, errorCode}, 'Certificate Validated')
+    }
+    callback(-3) // use chrome validation
+  })
+
+  app.on('select-client-certificate', (event, webContents, url, list, callback) => {
+    log.debug({url}, 'select-client-certificate')
+    callback()
+  })
+
+  app.on('certificate-error', (event, webContents, url, error, certificate, callback) => {
+    log.warn({url, error, certificate: printCertificate(certificate)}, 'App Certificate Error')
+    callback(false)
```

Comment:
Why `false`?

Reply:
Need to callback(false) to not accept any certificate.